test/e2e/framework/pv_util.go

```diff
@@ -901,6 +911,44 @@ func CreateSecPodWithNodeName(client clientset.Interface, namespace string, pvcl
 	return pod, nil
 }
 
+// SetNodeAffinityRequirement sets affinity with specified operator to nodeName to nodeSelection
+func SetNodeAffinityRequirement(nodeSelection *NodeSelection, operator v1.NodeSelectorOperator, nodeName string) {
+	// Add node-anti-affinity.
+	if nodeSelection.Affinity == nil {
+		nodeSelection.Affinity = &v1.Affinity{}
+	}
+	if nodeSelection.Affinity.NodeAffinity == nil {
+		nodeSelection.Affinity.NodeAffinity = &v1.NodeAffinity{}
+	}
+	if nodeSelection.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
+		nodeSelection.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = &v1.NodeSelector{}
+	}
+	nodeSelection.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms = append(nodeSelection.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms,
+		v1.NodeSelectorTerm{
+			// https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity warns
+			// that "the value of kubernetes.io/hostname may be the same as the Node name in some environments and a different value in other environments".
+			// So this might be cleaner:
+			// MatchFields: []v1.NodeSelectorRequirement{
+			// 	{Key: "name", Operator: v1.NodeSelectorOpNotIn, Values: []string{nodeName}},
+			// },
+			// However, "name", "Name", "ObjectMeta.Name" all got rejected with "not a valid field selector key".
+
+			MatchExpressions: []v1.NodeSelectorRequirement{
+				{Key: "kubernetes.io/hostname", Operator: operator, Values: []string{nodeName}},
```

Comment:
what happens if nodeName is ""? like if l.config.ClientNodeName is blank

Reply:
I would expect it to fail and not match anything. I'm surprised the tests are passing though?

Reply:
/hold 
to test this out

Reply:
```SetNodeAffinityRequirement``` is currently only used in a way like:
1. Deploy first pod,
2. Check the actual node name where first pod was deployed,
3. Create node affinity/anti-affinity for the node 

So, currently it is not used for the case that ```nodeName``` is blank, as a result, this bug hasn't been appeared in actual test.

Let me add codes in ```SetNodeAffinityRequirement``` to fail for ```nodeName``` blank case.

Reply:
I started to think that we don't need to add codes to force failure in this case, because this case should fail properly without this check. Do you agree or are there any actions needed for this?

Reply:
I think this is fine